addons/l10n_pe_website_sale/views/templates.xml

```diff
@@ -0,0 +1,120 @@
+                        <t t-out="city.name" />
+                    </option>
+                </t>
+            </select>
+        </div>
+
+        <!-- show district -->
+        <div
+            t-attf-class="mb-3 #{error.get('l10n_pe_district') and 'o_has_error' or ''} col-lg-6 div_district"
+            t-att-style="not city and 'display: none'">
+            <label class="col-form-label" for="l10n_pe_district">District</label>
+            <select
+                id="l10n_pe_district"
+                name="l10n_pe_district"
+                t-attf-class="form-select #{error.get('l10n_pe_district') and 'is-invalid' or ''}"
+                data-init="1">
+                <t t-foreach="city_districts" t-as="district">
+                    <option
+                        t-att-value="district.id"
+                        t-att-selected="district.id == ('l10n_pe_district' in checkout and checkout['l10n_pe_district'] != '' and int(checkout['l10n_pe_district']))">
+                        <t t-out="district.name" />
+                    </option>
+                </t>
+            </select>
+        </div>
+
+    </template>
+
+    <template id="address" inherit_id="website_sale.address">
+        <xpath expr="//input[@name='vat']/.." position="before">
+            <t t-if="mode[1] == 'billing'" position="inside">
+                <t t-if="res_company.country_id.code == 'PE'">
+                    <t t-set="partner" t-value="website_sale_order.partner_id" />
+                    <t t-call="l10n_pe_website_sale.partner_info" />
+                </t>
+            </t>
+        </xpath>
+        <label for="vat" position="attributes">
+            <attribute name="t-if">res_company.country_id.code != 'PE'</attribute>
+        </label>
+        <label for="vat" position="after">
+            <label t-if="res_company.country_id.code == 'PE'" class="col-form-label label-optional" for="vat">
+                Number
+            </label>
+        </label>
+        <xpath expr="//select[@name='state_id']/.." position="after">
+            <div class="w-100" />
+            <t t-if="res_company.country_id.code == 'PE'">
+                <t t-call="l10n_pe_website_sale.partner_address_info" />
+            </t>
+        </xpath>
+        <!-- Sets the country code for every country option -->
+        <xpath expr="//t[@t-foreach='countries']//option" position="attributes">
+            <attribute name="t-att-code">c.code</attribute>
+        </xpath>
+    </template>
+
+    <template id="website_layout" name="Website Layout Inherit" inherit_id="website.layout" priority="99">
+        <xpath expr="//html" position="before">
+            <t
+                t-set="html_data"
+                t-value="dict(html_data, **{'data-is-peruvian-company': '1' if res_company.country_id.code == 'PE' else '0' })" />
+        </xpath>
+    </template>
```

Comment:
Not needed anymore
```suggestion

```

Reply:
Thanks for the review @JulienVR 

I tested your latest changes and it's now working for both the Billing and Shipping addresses. Anything else you want to suggest?

Reply:
Hello @JulienVR 

I think this should be ready now, I tested on runbot for both the billing and shipping form and it looked good to me.